packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,138 @@ void main() {
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: null,
+            decoration: const InputDecoration(labelText: 'label', alignLabelWithHint: true),
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // isEmpty: true causes the label to be aligned with the input text if alignLabelWithHint is set to true
+    await tester.pumpAndSettle();
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 48.0);
+
+    // [FormField] value is initially null.
+    expect(stateKey.currentState.value, equals(null));
+    // [InputDecorator] isEmpty property is true.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(true));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('label'));
+    await tester.pumpAndSettle();
+
+    // Tap outside the [DropdownButton] to close dropdown menu without selecting an option.
+    await tester.tapAt(Offset.zero);
+    await tester.pumpAndSettle();
+
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 48.0);
+
+    // [FormField] value is still null.
+    expect(stateKey.currentState.value, equals(null));
+    // [InputDecorator] isEmpty property is still true.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(true));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('label'));
+    await tester.pumpAndSettle();
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+    // [InputDecorator] isEmpty property is now false.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(false));
+    expect(tester.getTopLeft(find.text('label')).dy, 12.0);
+    expect(tester.getTopLeft(find.text('label')).dy, 12.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 24.0);
+  });
+
+  testWidgets('DropdownButtonFormField InputDecorator label is aligned to centre of TextField when alignLabelWithHint is false', (WidgetTester tester) async {
```

Comment:
Prefer [US English](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#spell-words-in-identifiers-and-comments-correctly) for Flutter:
```suggestion
  testWidgets('DropdownButtonFormField InputDecorator label is aligned to center of TextField when alignLabelWithHint is false', (WidgetTester tester) async {
```